web/tasks.py

```diff
@@ -465,3 +465,17 @@ def export_node(
     print(
         f"Generated export file ({filesizeformat(len(file_contents))}) in {round(after-before,2)} seconds."
     )
+
+
+@task
+@setup_django
+def populate_ali_licensed(ctx):
```

Comment:
It might be good to for this task to be able to operate in batches or on subsets.

Reply:
I tried running in batches, I tried using Django `Paginator`, and also tried first creating a list of `to_be_updated_ids` and then running a `.filter(id__in=<to_be_updated_ids>).update(ali_licensed=True)` on that list, but they all performed similarly. I will reach out to folks to see if there might be other ideas. In my local, it takes about 6 minutes to iterate over 1.8 million records.

Reply:
For this it seems reasonable to do `for node in tqdm(ContentNode.objects.iterator()):`. `iterator` will iterate through the results instead of loading them all into ram, so the job will have O(1) ram usage instead of O(n). `tqdm` will print progress during the run.

It can definitely be nice to run updates in batches, and in a way that is resumable, but since it only takes 6 minutes and doesn't have to be run in maintenance mode I think this is fine.

I'd also recommend saving if it changes, rather than if it's True, so you can use it in the future when the get_ali_license_text() logic changes.